vignettes/tutorial2.Rmd

```diff
@@ -0,0 +1,447 @@
+```
+
+### Reading the ESM data
+
+The ESM temperature and precipitation fields should be in netCDF filea.  The function that reads it returns a "griddata" structure that contains the data, as well as some information about the grid, such as the latitude, longitude, and time coordinates and the vector of area weights needed to compute grid averages.
+
+```{r readdata}
+filenameT <- file.path(datadir, 'tas_annual_esm_rcp_r2i1p1_2006-2100.nc')
+griddataT <- read.general(filenameT, 
+                          varname='tas', latvar='lat_2', lonvar='lon_2',
+                          timevar='time')
+tgav <- griddataT$vardata %*% griddataT$globalop
+
+
+filenameP <- file.path(datadir, 'pr_annual_esm_rcp_r2i1p1_2006-2100.nc')
+griddataP <- read.general(filenameP, 
+                          varname='pr', latvar='lat', lonvar='lon',
+                          timevar='time')
+```
+Here, we have read in the temperature netCDF data and used the global average operator to compute a time series of global mean temperatures. We have also read in the precipitation netCDF data. These variables will be used as input to the functions that analyze the ESM data and produce the joint temperature and precipitation fields. 
+For the fldgen method to work, each ESM variable must be able to accept residuals of between infinity and -infinity. This is already the case for temperature and no transformation is needed. Because precipitation cannot be less than 0, either the generated residuals have to be constrained to avoid negative precipitation while preserving the ESM spatiotemporal statistics, or the method must operate on a transformation of precipitation that can accept residuals between -infinity and infinity. The latter is more straightforward, and so residuals are generated for log(precipitation) rather than precipitation. Any function that is continuous, invertible, and monotonic and results in a transformed variable supported on (-infinity, infinity) will preserve the ESM spatiotemporal statistical properties as desired. 
+
+```{r transform}
+
+tvarconvert_fcn <- NULL
+pvarconvert_fcn <- log
+
+    # make sure supported on -infinity to infinity, add some output data to
+    # facilitate the reversing later
+    if(!is.null(tvarconvert_fcn)){
+        griddataT$vardata_raw <- griddataT$vardata
+        griddataT$vardata <- tvarconvert_fcn(griddataT$vardata)
+    } else {
+        griddataT$vardata_raw <- NULL
+    }
+    griddataT$tvarconvert_fcn <- tvarconvert_fcn
+
+
+    if(!is.null(pvarconvert_fcn)){
+        griddataP$vardata_raw <- griddataP$vardata
+        griddataP$vardata <- pvarconvert_fcn(griddataP$vardata)
+    } else {
+        griddataP$vardata_raw <- NULL
+    }
+    griddataP$pvarconvert_fcn <- pvarconvert_fcn
+```
+
+```{r numbergrids}
+# the temperature and precipitation data must be on the same grid size:
+ncol(griddataT$vardata) ==  ncol(griddataT$vardata) 
+
+Ngrid <- as.numeric(ncol(griddataT$vardata))
+```
+
+
+### Analyzing the input data
```

Comment:
I like your writing in this section.  It's a good use of what Pinker calls "classic style".  It's a clear and effortless read.